pkg/occlient/occlient_test.go

```diff
@@ -1946,6 +1965,84 @@ func TestNewAppS2I(t *testing.T) {
 	}
 }
 
+func TestGetImageStream(t *testing.T) {
+	/*
+		imagev1.ImageStream{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "nodejs",
+							Namespace: "openshift",
+						},
+						Spec: imagev1.ImageStreamSpec{
+							Tags: []imagev1.TagReference{
+								{
+									Annotations: map[string]string{
+										"supports": "nodejs",
+										"tags":     "builder,nodejs",
+									},
+									From: &corev1.ObjectReference{
+										Kind: "ImageStreamTag",
+										Name: "latest",
+									},
+									Name: "latest",
+								},
+							},
+						},
+					}
+	*/
+	tests := []struct {
+		name      string
+		imageNS   string
+		imageName string
+		imageTag  string
+		wantErr   bool
+		want      *imagev1.ImageStream
+	}{
+		{
+			name:      "Case: Valid request for imagestream of latest version and not namespace qualified",
+			imageNS:   "",
+			imageName: "foo",
+			imageTag:  "latest",
+			want:      fakeImageStream("foo", "testing", []string{"latest"}),
+		},
+		{
+			name:      "Case: Valid explicit request for specific namespace qualified imagestream of specific version",
+			imageNS:   "openshift",
+			imageName: "foo",
+			imageTag:  "latest",
+			want:      fakeImageStream("foo", "openshift", []string{"latest", "3.5"}),
+		},
+		{
+			name:      "Case: Valid explicit request for specific imagestream of specific version qualfied from openshift namespace and not in current namespace",
+			imageNS:   "",
+			imageName: "foo",
+			imageTag:  "latest",
+			want:      fakeImageStream("foo", "openshift", []string{"latest", "3.5"}),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fkclient, fkclientset := FakeNew()
+			fkclient.namespace = "testing"
+
+			fkclientset.ImageClientset.PrependReactor("get", "imagestreams", func(action ktesting.Action) (bool, runtime.Object, error) {
+				return true, fakeImageStream(tt.imageName, "testing", []string{"latest"}), nil
+			})
+
+			fkclientset.ImageClientset.PrependReactor("get", "imagestreams", func(action ktesting.Action) (bool, runtime.Object, error) {
+				return true, fakeImageStream(tt.imageName, "openshift", []string{"latest", "3.5"}), nil
+			})
```

Comment:
@kadel It looks like the prepend reactor calls here behave randomly and return incorrect images to incorrect namespaces. Can you please help me fix it...